lib/const.rb

```diff
@@ -57,6 +57,27 @@
   HOME = CREW_PREFIX + Dir.home
 end
 
+CREW_IN_CONTAINER = File.exist?('/.dockerenv') || !ENV['CREW_IN_CONTAINER'].to_s.empty?
+
+CREW_CPU_VENDOR = CPUINFO['vendor_id'] || 'unknown'
+# vendor_id may not exist on non-x86 platforms.
+CREW_IS_AMD = ARCH == 'x86_64' ? CPUINFO['vendor_id'].include?('AuthenticAMD') : false
+CREW_IS_INTEL = ARCH == 'x86_64' || ARCH == 'i686' ? CPUINFO['vendor_id'].include?('GenuineIntel') : false
+
+# Use sane minimal defaults if in container and no override specified.
+if CREW_IN_CONTAINER && ENV['CREW_KERNEL_VERSION'].to_s.empty?
+  case ARCH
+  when 'i686'
+    CREW_KERNEL_VERSION = '3.8'
+  when 'aarch64', 'armv7l'
+    CREW_KERNEL_VERSION = '4.14'
+  when 'x86_64'
+    CREW_KERNEL_VERSION = '4.14'
+  end
```

Comment:
Why do we care what the kernel version is inside the container?  Shouldn't we only be concerned with actual hardware like Chromebooks instead?

Reply:
Exactly! The kernel detection code here overrides kernel version detection when we are in a container precisely because we don't care what the kernel version in the container is since that reports the container's host kernel version.